activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -942,6 +942,15 @@ def configure_connection
 
           # ...and send them all in one query
           raw_execute("SET #{encoding} #{sql_mode_assignment} #{variable_assignments}", "SCHEMA")
+
+          if @config[:isolation_level]
+            # Sending as a separate query since different spots that might intercept the query (Vitess, proxysql)
+            # may rely on SET SESSION TRANSACTION ISOLATION LEVEL as a single statement
```

Comment:
Is this really a concern? Because it means one extra roundtrip which isn't negligible.